chain/messagepool/pruning.go

```diff
@@ -0,0 +1,311 @@
+package messagepool
+
+import (
+	"bytes"
+	"context"
+	big2 "math/big"
+	"sort"
+	"time"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/lotus/build"
+	"github.com/filecoin-project/lotus/chain/messagepool/gasguess"
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/filecoin-project/lotus/chain/vm"
+	"github.com/filecoin-project/specs-actors/actors/abi"
+	"github.com/filecoin-project/specs-actors/actors/abi/big"
+	"github.com/ipfs/go-cid"
+	"golang.org/x/xerrors"
+)
+
+func (mp *MessagePool) pruneExcessMessages() error {
+	start := time.Now()
+	defer func() {
+		log.Infow("message pruning complete", "took", time.Since(start))
+	}()
+
+	mp.lk.Lock()
+	defer mp.lk.Unlock()
+
+	if mp.currentSize < mp.maxTxPoolSizeHi {
+		return nil
+	}
+
+	pruneCount := mp.currentSize - mp.maxTxPoolSizeLo
+
+	// Step 1. Remove all 'future' messages (those with a nonce gap)
+	npruned, err := mp.pruneFutureMessages()
+	if err != nil {
+		return xerrors.Errorf("failed to prune future messages: %w", err)
+	}
+
+	pruneCount -= npruned
+	if pruneCount <= 0 {
+		return nil
+	}
+
+	// Step 2. prune messages with the lowest gas prices
+
+	log.Warnf("still need to prune %d messages", pruneCount)
```

Comment:
wanna spell this out with "after pruning future messages, still need to prune %d..."

Reply:
um, also wanna actually call pruneMessages